compiler/rustc_lint/src/lints.rs

```diff
@@ -600,7 +600,7 @@ pub(crate) enum PtrNullChecksDiag<'a> {
         label: Span,
     },
     #[diag(lint_ptr_null_checks_fn_ret)]
-    FnRet { fn_name: Ident },
+    FnRet { fn_name: Symbol },
```

Comment:
If a function is named `async` in edition 2018 it should be reported as `r#async`, but symbols don't carry the "raw-ness" unlike ident.

Are we fine having inaccurate item name in some edge-cases ?

Reply:
oh good point

of course we don't have any tests for this lol